sharething_pwa/src/AddItem/AddItemScreen.tsx

```diff
@@ -3,6 +3,11 @@ import { withFirebase } from '../Firebase';
 import { Form, Button, FormControlProps, FormControl } from 'react-bootstrap';
 import Firebase from '../Firebase';
 
+const INITIAL_STATE = {
+  itemName: '' as string,
+  itemDescription: '' as string,
+  error: '' as string
+};
```

Comment:
1) `'' as string` is pointless. `''` is string, there is no need for casting. Casting is usually needed when you know more about a type than typescript knows, so you transfer that knowledge to TS by casting
2) put type on INITIAL_STATE object, not the properties. Like this:
```
const INITIAL_STATE: State = {...}
```